test/wasm2c/hello.txt

```diff
@@ -844,9 +844,24 @@ static const u8 data_segment_data_w2c_test_d0[] = {
   0x2e, 0x0a, 
 };
 
+static const u8* active_data_segment_data_ptrs[] = {
+  data_segment_data_w2c_test_d0,
+};
+
+static const size_t active_data_segment_sizes[] = {
+  14,  // data_segment_data_w2c_test_d0
+};
+
+static const u32 active_data_segment_offsets[] = {
+  8u,  // data_segment_data_w2c_test_d0
+};
+
 static void init_memories(w2c_test* instance) {
   wasm_rt_allocate_memory(&instance->w2c_memory, 1, 65536, 0, 65536);
-  LOAD_DATA(instance->w2c_memory, 8u, data_segment_data_w2c_test_d0, 14);
+  for (int i = 0; i < sizeof(active_data_segment_data_ptrs) / sizeof(active_data_segment_data_ptrs[0]); i++) {
+    
+    LOAD_DATA(instance->w2c_memory, active_data_segment_offsets[i], active_data_segment_data_ptrs[i], active_data_segment_sizes[i]);
+  }
```

Comment:
Do you know why the compiler was using more memory in the old form?  Is it because of inlining somehow? 

That amount of static data here is the same right?   Does the program you are having trouble with contains a huge about of segments?